Mathlib/Tactic/Bound.lean

```diff
@@ -0,0 +1,307 @@
+  · exact pow_le_pow_of_le_one a0 a1 mn
+
+/-- Branch on `1 ≤ x ∨ x ≤ 1` for `x^y` -/
+lemma Real.rpow_le_rpow_of_exponent_le_or_ge {x y z : ℝ}
+    (h : 1 ≤ x ∧ y ≤ z ∨ 0 < x ∧ x ≤ 1 ∧ z ≤ y) : x ^ y ≤ x ^ z := by
+  rcases h with ⟨x1, yz⟩ | ⟨x0, x1, zy⟩
+  · exact Real.rpow_le_rpow_of_exponent_le x1 yz
+  · exact Real.rpow_le_rpow_of_exponent_ge x0 x1 zy
+
+-- Register guessing rules
+attribute [bound]
+  -- Which side of the `max` should we use as the lower bound?
+  le_max_of_le_left_or_le_right
+  lt_max_of_lt_left_or_lt_right
+  -- Which side of the `min` should we use as the upper bound?
+  min_le_of_left_le_or_right_le
+  min_lt_of_left_lt_or_right_lt
+  -- Given `a^m ≤ a^n`, is `1 ≤ a` or `a ≤ 1`?
+  pow_le_pow_right_of_le_one_or_one_le
+  Real.rpow_le_rpow_of_exponent_le_or_ge
+
+end Guessing
+
+/-!
+### Closing tactics
+-/
+
+/-- Close numerical goals with `norm_num` -/
+def boundNormNum : Aesop.RuleTac :=
+  Aesop.SingleRuleTac.toRuleTac fun i => do
+    let tac := do NormNum.elabNormNum .missing .missing .missing
+    let goals ← Lean.Elab.Tactic.run i.goal tac |>.run'
+    if !goals.isEmpty then failure
+    return (#[], some (.ofTactic 1  `(tactic| norm_num)), some .hundred)
+attribute [aesop unsafe 10% tactic (rule_sets [Bound])] boundNormNum
+
+/-- Close numerical and other goals with `linarith` -/
+def boundLinarith : Aesop.RuleTac :=
+  Aesop.SingleRuleTac.toRuleTac fun i => do
+    Linarith.linarith false [] {} i.goal
+    return (#[], some (.ofTactic 1  `(tactic| linarith)), some .hundred)
+attribute [aesop unsafe 5% tactic (rule_sets [Bound])] boundLinarith
+
+/-!
+### `bound` tactic implementation
+-/
+
+/-- Pull the array out of an optional `"[" term,* "]"` syntax, or return `#[]` -/
+def maybeTerms : Syntax → Array Syntax
+  | node _ _ #[_, node _ _ s, _] => s.getEvenElems
+  | _ => #[]
+
+/-- Add extra hypotheses -/
+def addHyps (xs : Array Syntax) : TacticM Unit :=
+  if xs.isEmpty then pure () else Tactic.withMainContext do
+    for x in xs do
+      let v ← elabTerm x none
+      let t ← inferType v
+      liftMetaTactic fun g ↦ do
+        let g ← g.assert `h t v
+        let (_, g) ← g.intro1P
+        return [g]
+
+/-- Aesop configuration for `bound` -/
+def boundConfig : Aesop.Options := {
+  enableSimp := false
+}
+
+end Bound
+
+/-- `bound` tactic for proving inequalities via straightforward recursion on expression structure -/
+elab "bound" lemmas:(("[" term,* "]")?) : tactic => do
+  Bound.addHyps (Bound.maybeTerms lemmas)
+  let tac ← `(tactic| aesop (rule_sets [Bound, -default]) (config := Bound.boundConfig))
+  liftMetaTactic fun g ↦ do return (← Lean.Elab.runTactic g tac.raw).1
+
+/-- `bound`, but return a proof script.
+
+TODO: Add support for additional hypotheses via `bound? [h0, ...]`.  Currently these are not yet
+linked into Aesop's proof script generation. -/
+elab "bound?" : tactic => do
+  let tac ← `(tactic| aesop? (rule_sets [Bound, -default]) (config := Bound.boundConfig))
```

Comment:
```suggestion
  let tac ← `(tactic| aesop? (rule_sets := [Bound, -default]) (config := Bound.boundConfig))
```

Reply:
Done (resp. other places, and other aesop changes).